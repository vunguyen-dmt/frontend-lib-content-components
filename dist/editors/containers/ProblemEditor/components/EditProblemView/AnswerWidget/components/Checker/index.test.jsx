import { shallow } from 'enzyme';
import Checker from '.';

const props = {
  hasSingleAnswer: true,
  answer: {
    id: 'A',
    title: 'Answer 1',
    correct: true,
    selectedFeedback: 'some feedback',
  },
  setAnswer: jest.fn(),
};
describe('Checker component', () => {
  test('with single answer', () => {
    expect(shallow(<Checker {...props} />)).toMatchSnapshot();
  });

  test('with multiple answers', () => {
    expect(shallow(<Checker {...props} hasSingleAnswer={false} />)).toMatchSnapshot();
  });

  test('with disabled', () => {
    expect(shallow(<Checker {...props} disabled />)).toMatchSnapshot();
  });
});
