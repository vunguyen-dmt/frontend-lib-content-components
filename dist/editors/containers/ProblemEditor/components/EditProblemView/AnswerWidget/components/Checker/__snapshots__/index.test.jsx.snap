// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checker component with disabled 1`] = `
<Fragment>
  <Radio
    checked={true}
    className="pt-2.5"
    disabled={true}
    isValid={true}
    onChange={[Function]}
    value="A"
  />
  <Form.Label
    className="pt-2"
  >
    A
  </Form.Label>
</Fragment>
`;

exports[`Checker component with multiple answers 1`] = `
<Fragment>
  <Form.Checkbox
    checked={true}
    className="pt-2.5"
    disabled={false}
    isValid={true}
    onChange={[Function]}
    value="A"
  />
  <Form.Label
    className="pt-2"
  >
    A
  </Form.Label>
</Fragment>
`;

exports[`Checker component with single answer 1`] = `
<Fragment>
  <Radio
    checked={true}
    className="pt-2.5"
    disabled={false}
    isValid={true}
    onChange={[Function]}
    value="A"
  />
  <Form.Label
    className="pt-2"
  >
    A
  </Form.Label>
</Fragment>
`;
